app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,28 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
```

Comment:
This appears to me to be an unnecessary code because you are not using the @post anywhere in the 'new' method. Similarly, @user is unnecessary.